### PR TITLE
various naming conventions adjustments

### PR DESCRIPTION
**Pull request by stas00**
I started to work on convention/style tweaks discussed [here](http://forums.fast.ai/t/abbreviations-for-train-validation-test-variables/19618/43) with 001 plus updated abbr.md

And then some extras to validate with you @jph00:

1) in `dev_nb/001a_nn_basics.ipynb`:

Should we s/preds/log_preds/g to indicate that they are log()ed?

2) in `nb_001b.py`:

```
def listify(x=None, y=None):
def compose(funcs):
    def _inner(x, y=None):
```

we have x's here as well, which aren't necessarily torch, are they? So these should probably stay x regardless. it might be very difficult not to use x ;)

### CODE REVIEW

**Comment by stas00** [comment]
can I move to 002? I see you were still making changes to 002. Thanks.

**Comment by jph00** [comment]
> Should we s/preds/log_preds/g to indicate that they are log()ed?

Good idea.

> we have x's here as well, which aren't necessarily torch, are they? So these should probably stay x regardless. it might be very difficult not to use x ;)

Actually listify is a particularly bad example of using x & y I think. Maybe `p` and `q`? That's what `itertools` uses for arbitrary sequences: https://docs.python.org/2/library/itertools.html

Perhaps we should be using `t` for a arbitrary tensor instead of `x`? I do feel that the current notebooks and code are more confusing than they should be due to the current use of `x`.

Also, `dep` and `indep` seem too long for things that are so common, which already have standard one letter names (`x` and `y`).

> can I move to 002? I see you were still making changes to 002. Thanks.

Yup I just pushed some minor last changes. I'm starting on 003 now.

File: docs/abbr.md [hunk 1]
```diff
@@ -53,7 +54,7 @@ Note that there are always exceptions, especially when we try to comply with the
 |                  | input / output                      | io             |                                                  |
 |                  | object                              | obj            |                                                  |
 |                  | string                              | s              |                                                  |
-|                  | class                               | cls            |                                                  |
+|                  | class                               | cl             | cl, classes                                      |
```
Comment (jph00) [inline docs/abbr.md]: This should still be shown as `cls` for classmethod 1st param right?

File: docs/abbr.md [hunk 2]
```diff
@@ -33,10 +33,11 @@ Note that there are always exceptions, especially when we try to comply with the
 |                  | Convert between concepts            | 2              | name2idx(), label2idx(), seq2seq                 |
 | **Aggressive**   |                                     |                |                                                  |
 |                  | function                            | f              |                                                  |
-|                  | input                               | x              |                                                  |
+|                  | torch input                         | x              |                                                  |
 |                  | key                                 | k              |                                                  |
 |                  | value                               | v              |                                                  |
 |                  | index                               | i              |                                                  |
+|                  | lambda var                          | o              |                                                  |
```
Comment (jph00) [inline docs/abbr.md]: These two uses of `o` could be combined, e.g. "generic object argument e.g in list comprehensions and lambdas"

File: docs/abbr.md (hunk 1 above)
Comment (stas00) [reply docs/abbr.md]: You suggested cl, and classes here: 
http://forums.fast.ai/t/abbreviations-for-train-validation-test-variables/19618/31
we agreed that both cls and class shouldn't be used as they are reserved words.

File: docs/abbr.md (hunk 2 above)
Comment (stas00) [reply docs/abbr.md]: done

**Comment by stas00** [comment]
> Actually listify is a particularly bad example of using x & y I think. Maybe p and q? That's what itertools uses for arbitrary sequences: https://docs.python.org/2/library/itertools.html

sounds good to me. or `a` and `b` (except `b` is taken for batch)

> Perhaps we should be using t for a arbitrary tensor instead of x? I do feel that the current notebooks and code are more confusing than they should be due to the current use of x.

I'd like that.

should we then say `tx`, and `ty` (or `xt`/`xy`) for tensor x and tensor y, when it's a dep/indep tensors?

> Also, dep and indep seem too long for things that are so common, which already have standard one letter names (x and y).

we use dep for the np column and now suggested to use `dep_col` for that one.

but otherwise drop `dep`/`indep`?

File: docs/abbr.md (hunk 1 above)
Comment (jph00) [reply docs/abbr.md]: Sorry if I didn't explain properly; yes `cl` and `classes` are correct for that purpose, but `cls` is not a reserved word - it is simply the (by convention) name of the first parameter to a classmethod in python. I'm just wondering if we should explicitly mention this usage here.

**Comment by jph00** [comment]
> sounds good to me. or a and b (except b is taken for batch)

`p` and `q` don't have any conflicts, and are consistent with the python stdlib, so let's do that.

> should we then say tx, and ty (or xt/xy) for tensor x and tensor y, when it's a dep/indep tensors?

No, the type annotations (when we add them) will tell you the type. It doesn't need to be part of the name.

> we use dep for the np column and now suggested to use dep_col for that one.

I'm thinking we replace dep and indep with x and y entirely. So it would be `y_col`.

**Pull request merged**